### PR TITLE
logs: make log file world readable

### PR DESCRIPTION
**Pull request by CalvoM**
## Proposed Commit Message

logs: make log file world readable

Fixes: #1473 

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by renanrodrigo** [comment]
Thanks for the change, @CalvoM 
We have some tests failing though

**Comment by lucasmoura** [comment]
@CalvoM we just need to double check if `auto-attach` is not leaking any tokens when it succeeds or it fails.
We can use a PRO machine for that.

If we can confirm that, I see no problem with the PR

File: debian/ubuntu-advantage-tools.postinst [hunk 1]
```diff
@@ -421,14 +421,11 @@ case "$1" in
...
-      # We modify permissions for ubuntu-advantage.log because
-      # in a past version of UA, this log file was world readable.
-      # This isn't necessary for the timer/license-check log files,
-      # because they have always been only root-readable.
+      # /var/log/ubuntu-advantage.log needs to be world-readable
       if [ ! -f /var/log/ubuntu-advantage.log ]; then
           touch /var/log/ubuntu-advantage.log
       fi
-      chmod 0600 /var/log/ubuntu-advantage.log
+      chmod 0644 /var/log/ubuntu-advantage.log
```
Comment (orndorffgrant) [inline debian/ubuntu-advantage-tools.postinst:428]: We probably also need something like this for the other log files as well:
```bash
if [ -f /var/log/ubuntu-advantage-timer.log ]; then
  chmod 0644 /var/log/ubuntu-advantage-timer.log
fi
if [ -f /var/log/ubuntu-advantage-license-check.log ]; then
  chmod 0644 /var/log/ubuntu-advantage-license-check.log
fi
```

Comment (CalvoM) [reply debian/ubuntu-advantage-tools.postinst:428]: I have added the two, plus an extra one i.e. `/var/log/ubuntu-advantage-daemon.log`

**Comment by orndorffgrant** [comment]
I think this looks good but we should wait until after #1897 lands. I'm planning on doing a thorough audit of our logs for secrets in all scenarios as part of merging that PR.

**Review by orndorffgrant — APPROVED** [review]
This looks good and I wasn't able to replicate the jenkins failures.

Just needs a rebase to resolve conflicts and then we can merge

**Comment by orndorffgrant** [comment]
Jenkins failures are passing locally for me. Looks good!

File: debian/ubuntu-advantage-tools.postinst [hunk 2]
```diff
@@ -418,15 +418,24 @@ case "$1" in
...
+          chmod 0644 /var/log/ubuntu-advantage-timer.log
+      fi
+
+      if [ -f /var/log/ubuntu-advantage-license-check.log ]; then
+          chmod 0644 /var/log/ubuntu-advantage-license-check.log
+      fi
+
+      if [ -f /var/log/ubuntu-advantage-daemon.log ]; then
+          chmod 0644 /var/log/ubuntu-advantage-daemon.log
+      fi
```
Comment (lucasmoura) [inline debian/ubuntu-advantage-tools.postinst:438]: I think we can use a loop structure here

Comment (orndorffgrant) [reply debian/ubuntu-advantage-tools.postinst:438]: @CalvoM reminder that I think we want to make this small change before merging this. Something like:

```bash
for log_suffix in timer license-check daemon
do
    if [ -f /var/log/ubuntu-advantage-${log_suffix}.log ]; then
        chmod 0644 /var/log/ubuntu-advantage-${log_suffix}.log
    fi
done
```

CC @lucasmoura this is what you were imaging right?

Comment (lucasmoura) [reply debian/ubuntu-advantage-tools.postinst:438]: @orndorffgrant yes

Comment (CalvoM) [reply debian/ubuntu-advantage-tools.postinst:438]: Okay, time to learn bash properly :-)

File: debian/ubuntu-advantage-tools.postinst (hunk 1 above)
Comment (lucasmoura) [reply debian/ubuntu-advantage-tools.postinst:425]: @orndorffgrant @renanrodrigo do we want to perform any additional testing before making the `ubuntu-advantage.log` world readable ?

**Review by lucasmoura — APPROVED** [review]
The change looks good to me. I just to confirm if we should run any additional tests here for the main log

**Comment by orndorffgrant** [comment]
@lucasmoura I did what I think was a pretty thorough test when reviewing #1897, so I'm +1. Unfortunately I didn't document my tests :/. I basically did every cli action I could think of and manually looked through the logs and couldn't find any leaked secrets.

It'd probably be a good idea for either you or @renanrodrigo to also test any scenarios you can think of before we land this

**Comment by lucasmoura** [comment]
I have used this script to test this change:
```sh
series=$1

# Make sure deb is in the same directory as this script
deb=$2
name=$series-log

multipass delete $name
multipass purge
multipass launch $series --name $name
sleep 10
multipass transfer $deb $name:/tmp/ua.deb

multipass exec $name -- sudo dpkg -i /tmp/ua.deb
multipass exec $name -- ua version
multipass exec $name -- sudo ua attach $UACLIENT_BEHAVE_CONTRACT_TOKEN
multipass exec $name -- sudo ua refresh
multipass exec $name -- sudo rm /var/lib/ubuntu-advantage/private/machine-token.json
multipass exec $name -- sudo ua attach $UACLIENT_BEHAVE_CONTRACT_TOKEN
multipass exec $name -- sudo ua enable cis --assume-yes
multipass exec $name -- sudo ua disable esm-infra --assume-yes
multipass exec $name -- sudo ua status
multipass exec $name -- sudo ua detach --assume-yes
multipass exec $name -- sudo ua attach $UACLIENT_BEHAVE_CONTRACT_TOKEN
```

@renanrodrigo @orndorffgrant 
Let me know if you think we need to cover more scenarios

**Review by lucasmoura — APPROVED** [review]

**Comment by orndorffgrant** [comment]
@lucasmoura That looks good to me. I'm okay merging this PR :+1:

**Pull request merged**